src/zarr/core/array.py

```diff
@@ -3448,4 +3625,442 @@ def _get_default_codecs(
     else:
         dtype_key = "numeric"
 
-    return [{"name": codec_id, "configuration": {}} for codec_id in default_codecs[dtype_key]]
+    return cast(list[dict[str, JSON]], default_codecs[dtype_key])
+
+
+FiltersParam: TypeAlias = (
+    Iterable[dict[str, JSON] | ArrayArrayCodec]
+    | ArrayArrayCodec
+    | Iterable[numcodecs.abc.Codec]
+    | numcodecs.abc.Codec
+    | Literal["auto"]
+    | None
+)
+CompressorsParam: TypeAlias = (
+    Iterable[dict[str, JSON] | BytesBytesCodec]
+    | BytesBytesCodec
+    | numcodecs.abc.Codec
+    | Literal["auto"]
+    | None
+)
+ArrayBytesCodecParam: TypeAlias = dict[str, JSON] | ArrayBytesCodec | Literal["auto"]
+
+
+class ShardsConfigParam(TypedDict):
+    shape: ChunkCoords
+    index_location: ShardingCodecIndexLocation | None
+
+
+ShardsParam: TypeAlias = ChunkCoords | ShardsConfigParam | Literal["auto"]
+
+
+async def create_array(
+    store: str | StoreLike,
+    *,
+    name: str | None = None,
+    shape: ShapeLike,
+    dtype: npt.DTypeLike,
+    chunks: ChunkCoords | Literal["auto"] = "auto",
+    shards: ShardsParam | None = None,
+    filters: FiltersParam | None = "auto",
+    compressors: CompressorsParam = "auto",
+    array_bytes_codec: ArrayBytesCodecParam = "auto",
```

Comment:
I think this parameter is important, but I think we can do better with the name to express a bit more what it does, and detach the name from the type (which might change). Unfortunately the only ideas I have are `array_serializer` `chunk_serializer` but I'm sure there are better options.

Reply:
What about just `codec`?

Reply:
to me, `codec` is a category that includes `ArrayBytesCodec` and `BytesBytesCodec`; I think the ideal name would somehow convey that this entity is tasked with unpacking (or packing?) an array into bytes. `chunk_<verb>+er` seems like a good template, but I'm not sure what verbs fit well here. 🤔  `chunk_serializer`, `chunk_packer`, `chunk_streamer`... `chunk_serializer` seems the best to me but it's far from great.

Reply:
I think `chunk_serializer` is fine.
Maybe even just `serializer`. That would be more in line with `filters` and `compressors.`

Reply:
+1 to dropping `chunk_` as long as we don't think people will be confused about the scope